src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -239,12 +274,51 @@ private FileStream GetNewStream(long maxFileSize)
                                nextLogNumber.ToString("000", CultureInfo.InvariantCulture) + ".log";
                 if (File.Exists(fileName) && new FileInfo(fileName).Length >= maxFileSize)
                     continue;
+
+                KeepLogSize(logFiles, logGzFiles);
                 var fileStream = SafeFileStream.Create(fileName, FileMode.Append, FileAccess.Write, FileShare.Read, 32 * 1024, false);
                 fileStream.Write(_headerRow, 0, _headerRow.Length);
                 return fileStream;
             }
         }
 
+        private void KeepLogSize(string[] logFiles, string[] logGzFiles)
+        {
+            var logFilesInfo = logFiles.Select(f => new FileInfo(f));
+            var logGzFilesInfo = logGzFiles.Select(f => new FileInfo(f));
+            var totalLogSize = logFilesInfo.Sum(i => i.Length) + logGzFilesInfo.Sum(i => i.Length);
+
+            foreach (var log in logGzFilesInfo.Reverse())
+            {
+                if (totalLogSize > RetentionSize)
+                    try
+                    {
+                        log.Delete();
+                    }
+                    catch (Exception e)
+                    {
+                        //TODO What can we do in this situation??
```

Comment:
How can we handle a situation when can't be deleted?
in case of:
1) The file is in use
2) The remote connection to file lost

Reply:
We can usually state that losing I/O connectivity is _bad_. In this case, we can ignore the error and not handle it at all.
It will be fixed the next time we deal with it.

Reply:
and what about when the file in use? should I continue deleting the next log?
To delete until we rich the limit of log size or it should be considered as we deleted it?

Reply:
Ignore it, we'll be retrying this anyway